### PR TITLE
20117 Update migration scripts to store aliases in alternate names table

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#20117

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/scripts/manual_db_scripts/legal_name_change/transfer_to_new_lear.sql [hunk 1]
```diff
@@ -406,11 +406,11 @@ where sq.version != mv.max_version;
...
+-- aliases -> alternate_names
+transfer public.alternate_names from lear_old using
 SELECT a.id,
-       a.alias,
-       a.type,
+       a.alias                 as name,
+       a.type                  as name_type,
        a.business_id           as legal_entity_id,
        f.id                    as change_filing_id,
        COALESCE(av.version, 0) as version
```
Comment (argush3) [inline legal-api/scripts/manual_db_scripts/legal_name_change/transfer_to_new_lear.sql:416]: `start_date` in `alternate_names` needs to be populated

Comment (kzdev420) [reply legal-api/scripts/manual_db_scripts/legal_name_change/transfer_to_new_lear.sql:416]: What kind of value we use for start_date?

Comment (chenhongjing) [reply legal-api/scripts/manual_db_scripts/legal_name_change/transfer_to_new_lear.sql:416]: > `start_date` in `alternate_names` needs to be populated

What about `end_date` in `alterante_names`?

Comment (chenhongjing) [reply legal-api/scripts/manual_db_scripts/legal_name_change/transfer_to_new_lear.sql:416]: > What kind of value we use for start_date?

Think it's `filing.effective_date` for corresponding filing as per my previous conversation with Argus

Comment (argush3) [reply legal-api/scripts/manual_db_scripts/legal_name_change/transfer_to_new_lear.sql:416]: `end_date` is not required.

From what I can see, there are no deletions for `aliases`.  Only updates.

Comment (kzdev420) [reply legal-api/scripts/manual_db_scripts/legal_name_change/transfer_to_new_lear.sql]: Added the `start_date`

Comment (argush3) [reply legal-api/scripts/manual_db_scripts/legal_name_change/transfer_to_new_lear.sql]: it needs to be populated in history table too

Comment (kzdev420) [reply legal-api/scripts/manual_db_scripts/legal_name_change/transfer_to_new_lear.sql]: I did it

**Comment by argush3** [comment]
Did you verify that all aliases in the dev dump made it over to alternate names?

And can you also verify a something like the following business has all the historical entries listed below?  Minus the one with a null `end_transaction_id`

image

**Comment by kzdev420** [comment]
> Did you verify that all aliases in the dev dump made it over to `alternate_names`/`alternate_names_history` in the target db?
> 
> And can you also verify something like the following business has all the historical entries listed below? Minus the one with a null `end_transaction_id`
> 
> image

Screenshot 2024-03-14 at 2 20 01 PM
Screenshot 2024-03-14 at 2 20 44 PM

**Review by argush3 — APPROVED** [review]

**Review by JazzarKarim — APPROVED** [review]

**Review by chenhongjing — APPROVED** [review]

**Pull request merged**